### DISCUSSION TITLE
Pulling data from NCBI Nucleotide database

### DISCUSSION THREAD

**Discussion by yaaminiv** [RobertsLab/resources#1565] [Q&A] [answered]
I want transcriptome FASTA sequences from the NCBI nucleotide database (BioProject PRJNA74409, NCBI TSA accession numbers JR970414–JR999999 and JT000001–JT023377). Is there a programmatic way to grab all of these sequences at once? I found [this link that suggests using eutils](https://edwards.flinders.edu.au/ncbi-sequence-or-fasta-batch-download-using-entrez/), but I'd have to create a comma-separated list of 52963 accession numbers, and the [NCBI Eutils page](https://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EFetch) suggests using "the HTML POST method" if searching for more than 200 accessions (I also have no idea what they mean by HTML POST method). Any suggestions/insights on how to get these sequences easily?

**Comment by kubu4** [comment]
This notebook post might be helpful (it's very similar, but not exactly what you're doing):

https://robertslab.github.io/sams-notebook/2022/07/06/SRA-Data-S.namaycush-SRA-BioProject-PRJNA674328-Download-and-QC.html

Look it over and see if you have more questions and I'll be happy to help!

**Comment by kubu4** [comment, marked as answer]
Actually, you can ignore my previous post. Do the following:

1. Go to BioProject page.
2. Click on the "Number of Links" number of transcripts in the _Project Data_ table.

Then, send all of the info to FastA format:

![Screen cap showing how to select all data and download as FastA from NCBI](https://user-images.githubusercontent.com/4514104/211665287-2acee054-14a9-4c19-8c51-55a28bb9d6e0.png)

**Reply by kubu4** [reply]
That'll get you this FastA file:

```shell
head sequence.fasta 
>JT023377.1 TSA: Acropora millepora Cluster028086.Acmimixed mRNA sequence
AAGGTTCCAGAGCTTAATTCTATTGCGTTAGAAATTTTCCAGAAATGTATGTTAAATGGGATCACTATTG
ATGTGAATTGGATCCCAAGAGATTTCAACAGTGTGGCAGATGAGATTAGTAAGATAATAGATTACGATGA
TTACACAATTAATGATGATATTTTTGCTTTTTTGGACAAATCATGGGGACCCCATACAGTCGATCGTTTT
GCGTGTCACTATAATAAGAAGCTACCTTTATTTAATTCGAAGTTTTTTCAGCCAGGCACGAGTGGGGTGA
ATGCTTTCAGCCAAGACTGGGCTTTTGCCAATAATTGGTTATGTCCTCCCATTTATCTCACCGCGAGGGT
AGTTAATCATTTGAAAGCTTGTAGAGCGGCTGGGACCCTTATCGTCCCTTTGTGGAGATCGGCACATTTT
TGGCCAATTATTTGTGACGATGGAGTTCACTTTAGTAACTTTGTGCATGACTGGTATGTTTTGCCGCACA
TTCCTAATTTATTTATTAGAGGTAAAGCAAAGAATTGTATCTTCGGAAACGGCCAGTTGAAGTTTATAAT
GTTGGCATTAAGGATTGATTTTTCTGTCCCTCTTAGGTCCGCTGTACGAGGATTTTGTACTGAGTTTAAG
```

Count all the entries in the FastA file (i.e. all lines beginning with `>`) to make sure we got them all:

```shell
wc -l sequence.fasta
52963
```

**Comment by yaaminiv** [comment]
@kubu4 Thank you! If only the NCBI website wasn't a dumpster fire I probably would have figured it out myself :joy:
